wfe/wfe.go

```diff
@@ -1450,22 +1477,57 @@ func (wfe *WebFrontEndImpl) FinalizeOrder(
 		return
 	}
 
+	// spliting order identifiers per types
+	var orderDNSs []string
+	var orderIPs []net.IP
+	for _, ident := range orderIdentifiers {
+		switch ident.Type {
+		case acme.IdentifierDNS:
+			orderDNSs = append(orderDNSs, ident.Value)
+		case acme.IdentifierIP:
+			orderIPs = append(orderIPs, net.ParseIP(ident.Value))
+		default:
+			wfe.sendError(acme.MalformedProblem(
+				fmt.Sprintf("Order includes unknown identifier type %s", ident.Type)), response)
+		}
+	}
+	// looks like this spliting doesn't preserve order or Identifiers, so sort them again.
```

Comment:
I'm pretty sure the way you wrote the split code *does* preserve order. Can you produce a concrete example where it doesn't?

Reply:
wording was wrong:  looks like saving order to database breaks order of identifiers, for exemple https://travis-ci.org/letsencrypt/pebble/builds/537102621?utm_source=github_status&utm_medium=notification